src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -816,7 +816,7 @@ private async Task CommitCoreAsync(IUIThreadOperationContext operationContext, b
             var eventName = previewChanges ? FunctionId.Rename_CommitCoreWithPreview : FunctionId.Rename_CommitCore;
             using (Logger.LogBlock(eventName, KeyValueLogMessage.Create(LogType.UserAction), cancellationToken))
             {
-                var info = await _conflictResolutionTask.JoinAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
📝 This change was allowing the `_dismissed` field to be set from a background thread. Since `RenameService.ActiveSession` is cleared after `_dismissed` is set, it was allowing a UI thread check here to proceed between the point where `_dismissed` was set and the point where `ActiveSession` was cleared:

https://github.com/dotnet/roslyn/blob/4f0feeb9761ab2f070464e5db728d630f4bba1c4/src/EditorFeatures/Core/InlineRename/CommandHandlers/AbstractRenameCommandHandler_ReturnHandler.cs#L19